### PR TITLE
Add pod delete controls

### PR DESCRIPTION
**Pull request by tomwilkie**
Didn't add service, as it looks like [kubectl is responsible to tidying up pods...](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/delete.go)

### CODE REVIEW

**Comment by tomwilkie** [comment]
@paulbellamy do you think we need to the filter/state=deleted trick here so pods go away instantly?  Its a lot more work that with docker.

**Comment by paulbellamy** [comment]
We should. If we do it on containers, but not pods, then pods just seem broken.

File: report/id.go [hunk 1]
```diff
@@ -175,6 +175,15 @@ func ParsePodNodeID(podNodeID string) (namespaceID, podID string, ok bool) {
...
 
+// ParseServiceNodeID produces the namespace ID and service ID from an service node ID.
+func ParseServiceNodeID(serviceNodeID string) (namespaceID, serviceID string, ok bool) {
+	fields := strings.SplitN(serviceNodeID, ScopeDelim, 2)
+	if len(fields) != 2 {
+		return "", "", false
+	}
+	return fields[0], fields[1], true
+}
+
```
Comment (paulbellamy) [inline report/id.go]: This is probably just in for consistency, but it's not actually used anywhere...

**Comment by tomwilkie** [comment]
@paulbellamy I've added in the state / tombstone / shortcut reports stuff, and it kinda works.  Problem is multiple shortcut reports get generated very quickly (as when you delete pods, the RC recreates them very quickly), and this causes reports to get dropped.  I think we need to either add a buffer for reports, or start coalescing reports we might otherwise drop.  Thoughts?

**Comment by paulbellamy** [comment]
merging instead of dropping sounds reasonable.

**Comment by tomwilkie** [comment]
@paulbellamy I think this is ready to go in now.

File: probe/appclient/app_client.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func NewAppClient(pc ProbeConfig, hostname, target string, control xfer.ControlH
 			TLSClientConfig: httpTransport.TLSClientConfig,
 		},
 		conns:   map[string]xfer.Websocket{},
-		readers: make(chan io.Reader),
```
Comment (paulbellamy) [inline probe/appclient/app_client.go]: why 2?

File: render/pod.go [hunk 3]
```diff
@@ -15,7 +15,12 @@ const (
 
 // PodRenderer is a Renderer which produces a renderable kubernetes
 // graph by merging the container graph and the pods topology.
-var PodRenderer = FilterEmpty(report.Container,
+var PodRenderer = MakeFilter(
+	func(n report.Node) bool {
+		// Drop deleted containers
```
Comment (paulbellamy) [inline render/pod.go]: pods

File: probe/appclient/app_client.go (hunk 2 above)
Comment (tomwilkie) [inline probe/appclient/app_client.go]: Had to pick a number, and two allows for a pod deleted followed by a pod created event to be queued up.

File: report/id.go (hunk 1 above)
Comment (tomwilkie) [inline report/id.go]: Yeah just for consistency.  Gonna get used real soon I think.

Comment (paulbellamy) [inline report/id.go]: I hope not. let's add it when we need it.

**Pull request merged**